articles/quickstart/native/android/03-session-handling.md

```diff
@@ -74,133 +108,67 @@ private final AuthCallback callback = new AuthCallback() {
 :::
 
-Create an `AuthenticationAPIClient` instance:
-
-```java
-// app/src/main/java/com/auth0/samples/MainActivity.java
-
-AuthenticationAPIClient aClient = new AuthenticationAPIClient(auth0);
-```
-
-Use the refresh token to get new credentials:
-
-```java
-// app/src/main/java/com/auth0/samples/MainActivity.java
-
-String refreshToken = CredentialsManager.getCredentials(this).getRefreshToken();
-aClient.renewAuth(refreshToken)
-      .start(new BaseCallback<Credentials, AuthenticationException>() {
-
-          @Override
-          public void onSuccess(Credentials payload) {
-            String accessToken = payload.getAccessToken(); // New Access Token
-            String idToken = payload.getIdToken(); // New ID Token
-            //Save the new values
-          }
-
-          @Override
-          public void onFailure(AuthenticationException error) {
-            //show error
-          }
-      });
-```
 
 ## Log the User Out
 
-To log the user out, you must remove their credentials and navigate them to the login screen.
+To log the user out, you remove their credentials and navigate them to the login screen. When using a Credentials Manager you do that calling `clearCredentials`.
 
-For example, you can do the following:
+In the sample, the LoginActivity checks that a boolean extra is present in the Intent at the Activity launch, so that if this flag is true the credentials are first removed from the Credentials Manager.
 
 ```java
 // app/src/main/java/com/auth0/samples/MainActivity.java
 
 private void logout() {
-  CredentialsManager.deleteCredentials(this);
-  startActivity(new Intent(this, LoginActivity.class));
-  finish();
+    Intent intent = new Intent(this, LoginActivity.class);
+    intent.putExtra(LoginActivity.KEY_CLEAR_CREDENTIALS, true);
+    startActivity(intent);
+    finish();
+}
+
+// app/src/main/java/com/auth0/samples/LoginActivity.java
+@Override
+protected void onCreate(Bundle savedInstanceState) {
+    //...
+    if (getIntent().getBooleanExtra(KEY_CLEAR_CREDENTIALS, false)) {
+        credentialsManager.clearCredentials();
+    }
 }
 ```
 
 ::: note
-Depending on the way you store users' credentials, you delete them differently. 
-:::
-
-## Optional: Encapsulate Session Handling
-
-You can simplify the way you handle user sessions by storing token-related information and processes in a class. The class separates the logic for handling user sessions from the activity. 
-
-We recommend that you download the sample project from this tutorial and look at its implementation. Focus on the `CredentialsManager` class, which manages session handling, obtains user credentials from the `SharedPreferences` file and saves them.
+If you are not using our Credentials Manager classes, depending on the way you store users' credentials you delete them differently. 
```

Comment:
~If you are not using our Credentials Manager classes, depending on the way you store users' credentials you delete them differently. ~
If you are not using our Credentials Manager classes, you are responsible for ensuring that the user's credentials have been removed.